galley/pkg/kube/source/source.go

```diff
@@ -26,11 +26,26 @@ import (
 	kube_meta "istio.io/istio/galley/pkg/metadata/kube"
 	"istio.io/istio/galley/pkg/runtime"
 	"istio.io/istio/galley/pkg/runtime/resource"
+	"istio.io/istio/pkg/features/galley"
 	"istio.io/istio/pkg/log"
 )
 
 var scope = log.RegisterScope("kube", "kube-specific debugging", 0)
 
+func enabledKubeTypes() []kube.ResourceSpec {
+	allTypes := kube_meta.Types.All()
+	if galley.ConvertK8SService {
```

Comment:
Please don't use globals. This should be precalculated and passed as a parameter from the service entry point.